core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -98,7 +111,7 @@ class AdminManager(val config: KafkaConfig,
       val results = metadata.map { createTopicMetadata =>
         // ignore topics that already have errors
         if (createTopicMetadata.error == Errors.NONE) {
```

Comment:
Haha, I was trying to diagnose some test failures and reached the exact same conclusion at about the same time. There are quite a few places with the same issue. Will push an update ASAP.